src/transformers/models/llava/configuration_llava.py

```diff
@@ -116,15 +119,22 @@ def __init__(
                 vocab_size=32000,
                 projection_dim=768,
             )
-        self.vocab_size = self.vocab_size
 
-        self.text_config = text_config
+        self.vision_config = vision_config
 
-        if isinstance(self.text_config, dict):
+        if isinstance(text_config, dict):
             text_config["model_type"] = text_config["model_type"] if "model_type" in text_config else "llama"
-            self.text_config = CONFIG_MAPPING[text_config["model_type"]](**text_config)
-            self.vocab_size = self.text_config.vocab_size
+            text_config = CONFIG_MAPPING[text_config["model_type"]](**text_config)
         elif text_config is None:
-            self.text_config = CONFIG_MAPPING["llama"]()
+            text_config = CONFIG_MAPPING["llama"]()
+            if "vocab_size" in kwargs:
+                warnings.warn(
+                    "The `vocab_size` argument is deprecated and will be removed in v4.42, since it can be inferred from the `text_config`.",
+                    FutureWarning,
+                )
+                # set the vocab_size
+                text_config.vocab_size = kwargs["vocab_size"]
```

Comment:
Same comments as for vip llava:
* Just have a warning at the start of the init
* Add some deprecation for `vocab_size`